packages/web3-eth-contract/src/contract.ts

```diff
@@ -27,35 +37,32 @@ import {
 } from './types';
 import { getEstimateGasParams, getEthTxCallParams, getSendTxParams } from './utils';
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type ContractBoundMethod<Method extends ContractMethod<any>> = (
```

Comment:
should move these contract types in `web3-eth-contract/src/types.ts`

- ContractBoundMethod
- ContractMethodsInterface
- ContractEventsInterface
- ContractEventEmitterInterface
- EventParameters

Reply:
Updated a few times. Rest o the types have circular dependency with `LogSubscription` file, so had to declare these here.